node/pkg/adminrpc/adminserver.go

```diff
@@ -577,6 +578,137 @@ func wormholeRelayerSetDefaultDeliveryProvider(req *nodev1.WormholeRelayerSetDef
+	// Here the main difficulty is that we need to convert the input strings
+	// into the appropriate types first, as the abi.Pack function expects
+	// the arguments to be of certain go types, and there doesn't seem to be a
+	// function in the library that parses a string into that type, so we write that here.
+	args := make([]interface{}, len(evmCall.Argument))
+
+	method := parsedAbi.Methods[selector.Name]
+	if len(args) != len(method.Inputs) {
+		return nil, fmt.Errorf("expected %d arguments, got %d", len(method.Inputs), len(args))
+	}
+
+	for i, arg := range method.Inputs {
+		res, err := parseEvmArg(arg, evmCall.Argument[i])
+		if err != nil {
+			return nil, err
+		}
+		args[i] = res
+	}
+
+	encodedCall, err := parsedAbi.Pack(selector.Name, args...)
+	if err != nil {
+		return nil, err
+	}
+
+	var governanceContract [32]byte
+	copy(governanceContract[:], ethcommon.HexToAddress(evmCall.GovernanceContract).Bytes())
+	var targetContract [32]byte
+	copy(targetContract[:], ethcommon.HexToAddress(evmCall.TargetContract).Bytes())
+
+	v := vaa.CreateGovernanceVAA(timestamp, nonce, sequence, guardianSetIndex,
+		vaa.BodyGeneralPurposeGovernanceEvm{
+			ChainID:            vaa.ChainID(evmCall.ChainId),
+			GovernanceContract: governanceContract,
+			TargetContract:     targetContract,
+			Payload:            encodedCall,
+		}.Serialize())
+
+	return v, nil
+}
+
+func parseEvmArg(arg abi.Argument, input string) (interface{}, error) {
+	switch arg.Type.T {
+	case abi.UintTy:
+		if arg.Type.Size <= 64 {
+			u, err := strconv.ParseUint(input, 10, arg.Type.Size)
+			if err != nil {
+				return nil, err
+			}
+			if arg.Type.Size == 64 {
+				return u, nil
+			} else if arg.Type.Size == 32 {
+				return uint32(u), nil
+			} else if arg.Type.Size == 16 {
+				return uint16(u), nil
+			} else if arg.Type.Size == 8 {
+				return uint8(u), nil
+			} else {
+				return nil, fmt.Errorf("unsupported uint size %d", arg.Type.Size)
+			}
+		} else {
+			b := big.NewInt(0)
+			b.SetString(input, 10)
+			return b, nil
+		}
+	case abi.BoolTy:
+		b, err := strconv.ParseBool(input)
+		if err != nil {
+			return nil, err
+		}
+		return b, nil
+	case abi.StringTy:
+		return input, nil
+	case abi.AddressTy:
+		if !ethcommon.IsHexAddress(input) {
+			return nil, fmt.Errorf("invalid address %s", input)
+		}
+		return ethcommon.HexToAddress(input), nil
+	case abi.BytesTy:
+		b, err := hex.DecodeString(input)
+		if err != nil {
+			return nil, err
+		}
+		return b, nil
+	case abi.FixedBytesTy:
+		bytes, err := hex.DecodeString(input)
+		if err != nil {
+			return nil, err
+		}
+		if len(bytes) != arg.Type.Size {
+			return nil, fmt.Errorf("invalid byte length %d, expected %d", len(bytes), arg.Type.Size)
+		}
+		switch arg.Type.Size {
+		case 20:
+			var b [20]byte
+			copy(b[:], bytes)
+			return b, nil
+		case 32:
+			var b [32]byte
+			copy(b[:], bytes)
+			return b, nil
+		// TODO: do we want more fixed byte sizes?
```

Comment:
I remember we've used `bytes4` in various places but don't think anywhere that's relevant for governance messages yet.